sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,574 @@
+            foreach (ParameterSyntax parameter in method.ParameterList.Parameters)
+            {
+                // If there's no attribute, we can assume that this parameter is not a binding
+                if (parameter.AttributeLists.Count == 0)
+                {
+                    continue;
+                }
+
+                IParameterSymbol? parameterSymbol = model.GetDeclaredSymbol(parameter);
+
+                if (parameterSymbol is null)
+                {
+                    throw new InvalidOperationException($"The symbol for the parameter '{nameof(parameter)}' could not be found");
+                }
+
+                // Assumption: All parameters in a function method thathave an attribute will have just one (the trigger or input/output binding attribute)
+                AttributeData attributeData = parameterSymbol.GetAttributes().First();
+
+                // Get the attribute method symbol and treat the constructor as a method (IMethodSymbol)
+                AttributeSyntax attributeSyntax = parameter.AttributeLists.First().Attributes.First();
+                IMethodSymbol? attribMethodSymbol = model.GetSymbolInfo(attributeSyntax).Symbol as IMethodSymbol;
+
+                // Check if the attribute constructor has any parameters
+                if (attribMethodSymbol is null || attribMethodSymbol?.Parameters is null)
+                {
+                    throw new InvalidOperationException($"The constructor of attribute with syntax '{nameof(attributeSyntax)}' is invalid");
+                }
+
+                // Get binding info as a dictionary with keys as the property name and value as the property value
+                IDictionary<string, object> attributeProperties = GetAttributeProperties(attribMethodSymbol, attributeData);
+
+                // Grab some required binding info properties
+                string attributeName = attributeData.AttributeClass!.Name;
+                string bindingName = parameter.Identifier.ValueText;
+
+                // properly format binding types by removing "Attribute" and "Input" descriptors
+                string bindingType = attributeName.Replace("Attribute", "");
+                bindingType = bindingType.Replace("Input", "");
+
+                // Set binding direction
+                string bindingDirection = "In";
+                if (parameterSymbol.Type is INamedTypeSymbol parameterNamedType &&
+                    parameterNamedType.IsGenericType &&
+                    parameterNamedType.ConstructUnboundGenericType().ToString() == "Microsoft.Azure.Functions.DotNetWorker.OutputBinding<>")
+                {
+                    bindingDirection = "Out";
+                }
+
+                string? dataType = null;
+
+                // Check if parameter datatype is string or binary
+                // Is string parameter type
+                if (IsStringType(parameterSymbol.Type.MetadataName))
```

Comment:
Just discovered [this method](https://github.com/Azure/azure-functions-dotnet-worker/blob/95a10c7aa95da2ba1122201ef78f5059733fa283/sdk/Sdk.Generators/ISymbolExtensions.cs#L18) inside the generators sdk so will see if i can incorporate it. Would be better to use the full name.